spi-proposal.md

```diff
@@ -0,0 +1,249 @@
+5:  maxSpeed              (bits 0 - 6)
+6:  maxSpeed              (bits 7 - 14)
+7:  maxSpeed              (bits 15 - 21)
+8:  maxSpeed              (bits 22 - 28)
+9:  maxSpeed              (bits 29 - 32)
+10  csPin                 [optional] (0-127) The chip select pin number
+11: END_SYSEX
+```
+
+#### bitOrder
+```
+LSBFIRST = 0
+MSBFIRST = 1
+```
+
+#### dataMode
+
+| mode    | clock polarity (CPOL) | clock phase (CPHA) |
+| --------|-----------------------|--------------------|
+| 0       | 0                     | 0                  |
+| 1       | 0                     | 1                  |
+| 2       | 1                     | 0                  |
+| 3       | 1                     | 1                  |
+
+
+#### maxSpeed
+
+The maximum speed of communication with the SPI device. For a SPI device rated up to
+5 MHz, use 5000000.
+
+
+### SPI_END_TRANSACTION
+
+Signals the end of a transaction. Must be sent before starting communication
+with another SPI device.
+
+```
+0:  START_SYSEX
+1:  SPI_DATA              (0x68)
+2:  SPI_END_TRANSACTION   (0x02)
+3:  channel               (HW supports multiple SPI ports. range = 0-3, default = 0)
+4:  END_SYSEX
+```
+
+### SPI_TRANSFER
+
+Write, read or read and write one or more bytes.
+
+The user can continue sending as many transfer messages as necessary, even
+alternating between read-only, write-only and read-write as necessary within
+a single SPI transaction.
+
+When setting `0` (read/write) as the `transferOption`, a byte must be written for
+every byte received. The user must then extract data from the array of bytes returned
+in the `SPI_REPLY` message.
+
+If a CS pin is specified in the `SPI_BEGIN_TRANSACTION` message, it will be used
+according to the behavior specified by `pinOptions` (byte 4). This only covers
+common cases, with the default being set CS to active LOW. Using the `pinOptions`
+bits, it also enables the user to set the CS pin value only on start (bit 3) or
+only on end (bit 4) or ignore entirely (bit 2) in order to send data that requires
+multiple `SPI_TRANSFER` messages. Another option (bit 6) allows the user to
+specify whether or the CS pin should be toggled at the end of the message
+(typical SPI behavior) or between each byte sent (edge case).
+
+
+```
+0:  START_SYSEX
+1:  SPI_DATA              (0x68)
+2:  SPI_TRANSFER          (0x03)
+3:  channel               (HW supports multiple SPI ports. range = 0-3, default = 0)
+4:  pinOptions | transferOptions
+                          bits 0-1: transferOptions
+                            0 = read/write,
+                            1 = read-only,
+                            2 = write-only
+                            (see details in transferOptions section below)
+                          bits 2-4 : pinOptions
+                            bit 2: CS_DISABLE
+                            bit 3: CS_START_ONLY
+                            bit 4: CS_END_ONLY
+                            bit 5: CS_ACTIVE_EDGE
```

Comment:
`CS_TOGGLE` could possible be moved to `SPI_BEGIN_TRANSACTION` as well since I assume a SPI device that needs that behavior would need it for every transfer.